### DISCUSSION TITLE
Help with functionality / features

### DISCUSSION THREAD

**Discussion by xnmrph440** [abraunegg/onedrive#1973] [Q&A] [answered]
Hello all - apologies I'm new to Github and - relatively - new to Ubuntu although I'm beginning to slowly migrate my daily stuff to it and finding all the wee things I like need some extra work; especially with Microsoft things. 

@abraunegg - thanks for your help and guidance on the other place! I have 2 main questions:

I'm happy to open a new issue in the correct area if req'd. 

I have a very simple setup. I have the client configured to sync only one directory to OneDrive, I'll expand as I go but no need to move almost a terabyte to this machine. 

1. **Solved** - Logging - I have tried to change the config and set 'enable-logging' = true however when I re-run the sync I get the same response from azure as I did when I first set it up, where it asks for the response URL. Knowing I'd done this already then I reverted this and the sync was fine again, albeit without logging. I strongly suspect this is due to me following the incorrect install instructions and need to run through the steps as outlined in this previous discussion: https://github.com/abraunegg/onedrive/discussions/1772

2. Startup / Shutdown execution. I have achieved this already on startup by using the startup application loader built into Ubuntu and configured it with the correct command. This works perfectly. I can't achieve the same on shutdown. 

I have created a script in /etc/init.d/K99onedrive and made it executable. I then created a symlink within /etc/rc6.d/ however this doesn't run on power-off. I know this is something willy and my lack of experience is the cause. /6 

Edit: I tried to run the app from the rc6.d directory and it asks for the same response URL so i suspect it thinks it's a new install on both "issues" 

`onedrive --synchronize --single-directory 'directory-name'`

![Screenshot from 2022-05-29 21-45-27](https://user-images.githubusercontent.com/93046502/170890920-e7067987-a5f7-4480-8bba-f8bc98d22f2d.jpg)

**Comment by xnmrph440** [comment]
Edit - i have solved the logging issue - permissions related.

**Comment by abraunegg** [comment, marked as answer]
@xnmrph440 
> I strongly suspect this is due to me following the incorrect install instructions
> ...
> Edit - i have solved the logging issue - permissions related.

Glad you got your permissions issue fixed.

Please make sure when installing the client on Ubuntu you follow these instructions - https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md - and *only* these instructions ... all other instructions on the Internet will lead you to failure.

Next read and understand https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md#known-issues-with-installing-from-the-above-packages .. especially the first point .. that is a packaging issue that needs to be fixed .. but you need to take proactive action and remove the erroneously created file .. otherwise you will get into a world of hurt with other things.

> Startup / Shutdown execution. I have achieved this already on startup by using the startup application loader built into Ubuntu and configured it with the correct command. This works perfectly. I can't achieve the same on shutdown.

This is 100% the incorrect approach. Additionally, `--synchronize` is a one time sync, then will exit .. unsure if this is what you want. The client will also be running as 'root' - which, is why it is prompting you to re-auth - because 'root' did not auth the client originally ... 

What you need to do is enable the systemd service - the correct process is:
* Without notifications to GUI - https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#onedrive-service-running-as-a-non-root-user-via-systemd-all-linux-distributions
* With notifications to GUI - https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#onedrive-service-running-as-a-non-root-user-via-systemd-with-notifications-enabled-arch-ubuntu-debian-opensuse-fedora

This will ensure that the client is correctly started at system startup and the client shutdown when you stop your system.

If it is a 'one-time' sync you want, the systemd service at startup is still what you need to use - you just need to edit the systemd file and reconfigure it so that it uses `--synchronize` rather than `--monitor` , and edit the options so that it does not continuously try and restart after exiting.

Lastly - before enabling systemd - test everything to ensure it is doing what you want it to do. Use `--dry-run` to test.
